compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -626,7 +626,7 @@ impl<'a> Linker for GccLinker<'a> {
                 // it does support --strip-all as a compatibility alias for -s.
                 // The --strip-debug case is handled by running an external
                 // `strip` utility as a separate step after linking.
-                if self.sess.target.os != "illumos" {
+                if self.sess.target.os != "illumos" && self.sess.target.os != "solaris" {
```

Comment:
```suggestion
                if !self.sess.target.is_like_solaris {
```
This flag is generally preferred to checking for a specific OS, it probably makes sense to audit other uses of "illumos" and "solaris" in the compiler.